### PR TITLE
Renaming and adding specific formulation classes

### PR DESCRIPTION
**Pull request by carldlaird**

### CODE REVIEW

File: tests/neuralnet/test_nn_formulation.py [hunk 1]
```diff
@@ -48,32 +50,39 @@ def two_node_network(activation, input_value):
...
+    net, y = two_node_network('sigmoid', -2.0)
+    m.neural_net_block.build_formulation(FullSpaceSmoothNNFormulation(net))
+    assert m.nvariables() == 15
+    assert m.nconstraints() == 14
+
+    m.neural_net_block.inputs[0].fix(-2)
+    m.obj1 = pyo.Objective(expr=0)
+    status = pyo.SolverFactory("ipopt").solve(m, tee=True)
     assert abs(pyo.value(m.neural_net_block.outputs[0,0]) - y[0,0]) < 1e-6
     assert abs(pyo.value(m.neural_net_block.outputs[0,1]) - y[0,1]) < 1e-6
```
Comment (jalving) [inline tests/neuralnet/test_nn_formulation.py:79]: I think these are captured by the below tests. We could consider removing.

File: src/omlt/neuralnet/activations/__init__.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 ACTIVATION_FUNCTION_MAP = {
     "linear": linear_activation_function,
-#    "relu": bigm_relu_activation,
+#    "relu": relu_max_activation
     "sigmoid": sigmoid_activation_function,
     "softplus": softplus_activation_function,
     "tanh": tanh_activation_function
```
Comment (jalving) [inline src/omlt/neuralnet/activations/__init__.py:12]: Does `ACTIVATION_FUNCTION_MAP` need to be here? We import it as `_DEFAULT_ACTIVATION_FUNCTIONS` into `nn_formulation.py`, but then we define `_DEFAULT_ACTIVATION_CONSTRAINTS` separately in that file. Maybe these are attributes we want to expose to the user? If yes, should we put the constraint map here too?

File: src/omlt/neuralnet/nn_formulation.py [hunk 3]
```diff
@@ -6,7 +6,8 @@
 from omlt.neuralnet.layers.reduced_space import reduced_space_dense_layer
 from omlt.neuralnet.activations import (linear_activation_constraint,
                                         linear_activation_function,
-                                        bigm_relu_activation_constraint)
+                                        bigm_relu_activation_constraint,
+                                        ComplementarityReLUActivation)
```
Comment (jalving) [inline src/omlt/neuralnet/nn_formulation.py:10]: consider using `omlt.neuralnet.activations.linear` and `omlt.neuralnet.activations.relu` to be consistent with `omlt.neuralnet.activations.smooth` used below.

File: src/omlt/neuralnet/nn_formulation.py [hunk 4]
```diff
@@ -181,16 +254,23 @@ class ReducedSpaceNeuralNetworkFormulation(_PyomoFormulation):
     activation_constraints : dict-like or None
```
Comment (jalving) [inline src/omlt/neuralnet/nn_formulation.py:254]: todo: remove `layer_constraints` from doc string here

File: src/omlt/neuralnet/nn_formulation.py [hunk 5]
```diff
@@ -166,8 +177,70 @@ def output_assignment(b, *output_index):
...
+        """
+        super().__init__(network_structure)
+
+    def _supported_default_activation_constraints(self):
+        return {
+            "linear": linear_activation_constraint,
+            "sigmoid": sigmoid_activation_constraint,
+            "softplus": softplus_activation_constraint,
+            "tanh": tanh_activation_constraint
+        }
```
Comment (jalving) [inline src/omlt/neuralnet/nn_formulation.py:200]: Possible discussion item: query from `_DEFAULT_ACTIVATION_CONSTRAINTS` where we can. Might be easier to maintain when we add new default functions. we could just pop the 'relu'.

**Review by jalving — APPROVED** [review]
See comments. They can be addressed in separate issues.

File: tests/neuralnet/test_nn_formulation.py (hunk 1 above)
Comment (carldlaird) [reply tests/neuralnet/test_nn_formulation.py:79]: done.

File: src/omlt/neuralnet/nn_formulation.py (hunk 3 above)
Comment (carldlaird) [reply src/omlt/neuralnet/nn_formulation.py:10]: Went a different route and imported all from omlt.neuralnet.activations.

File: src/omlt/neuralnet/nn_formulation.py (hunk 4 above)
Comment (carldlaird) [reply src/omlt/neuralnet/nn_formulation.py:254]: doctoring fixed.

Comment (carldlaird) [reply src/omlt/neuralnet/nn_formulation.py:254]: Auto-correct: docstring fixed.

**Review by jalving — APPROVED** [review]

**Pull request merged**